flink-connectors/flink-orc/src/test/java/org/apache/flink/orc/OrcTableSourceITCase.java

```diff
@@ -0,0 +1,289 @@
+		);
+
+		tEnv.registerTableSource("orcTable", orc);
+
+		Table t = tEnv.scan("orcTable")
+			.select("list.at(2).get(0)");
+
+		DataSet<Row> dataSet = tEnv.toDataSet(t, Row.class);
+		List<Row> records = dataSet.collect();
+
+		Assert.assertEquals(records.get(0).toString(), "4");
+		Assert.assertEquals(records.get(1).toString(), "-100000");
+
+	}
+
+	@Test
+	public void testOrcMap() throws Exception {
+
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		BatchTableEnvironment tEnv = TableEnvironment.getTableEnvironment(env);
+
+		String path = orcPath + "TestOrcFile.test1.orc";
+		OrcTableSource orc = new OrcTableSource(
+			path,
+			"struct<boolean1:boolean,byte1:tinyint,short1:smallint,int1:int,long1:bigint,float1:float,double1:double,bytes1:binary,string1:string,middle:struct<list:array<struct<int1:int,string1:string>>>,list:array<struct<int1:int,string1:string>>,map:map<string,struct<int1:int,string1:string>>>"
+		);
+
+		tEnv.registerTableSource("orcTable", orc);
+
+		String query = "Select middle, list, map as mymapper from orcTable";
+		Table t = tEnv.sql(query);
+
+		tEnv.registerTable("mytable", t);
+		Table t1 = tEnv.sql("SELECT mymapper['chani'] FROM mytable");
+
+		DataSet<Row> dataSet = tEnv.toDataSet(t1, Row.class);
+		List<Row> records = dataSet.collect();
+
+		Assert.assertEquals(records.get(0).toString(), String.valueOf("null"));
+		Assert.assertEquals(records.get(1).toString(), "5,chani");
+
+	}
+
+	@Test
+	public void testOrcDecimal() throws Exception {
+
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		BatchTableEnvironment tEnv = TableEnvironment.getTableEnvironment(env);
+
+		String path = orcPath + "decimal.orc";
+		OrcTableSource orc = new OrcTableSource(
+			path,
+			"struct<_col0:decimal(10,5)>"
+		);
+
+		tEnv.registerTableSource("orcTable", orc);
+		Table t = tEnv.scan("orcTable")
+			.select("*")
+			.where("_col0 < 2000 && _col0 > 1991");
+
+		DataSet<Row> dataSet = tEnv.toDataSet(t, Row.class);
+		List<Row> records = dataSet.collect();
+
+		Assert.assertEquals(records.size(), 9);
+
+	}
+
+	@Test
+	public void testOrcLargeFile() throws Exception {
+
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		BatchTableEnvironment tEnv = TableEnvironment.getTableEnvironment(env);
+
+		String path = orcPath + "demo-11-none.orc";
+		OrcTableSource orc = new OrcTableSource(
+			path,
+			"struct<_col0:int,_col1:string,_col2:string,_col3:string,_col4:int,_col5:string,_col6:int,_col7:int,_col8:int>"
+		);
+
+		tEnv.registerTableSource("orcTable", orc);
+		Table t = tEnv.scan("orcTable")
+			.select("*");
+		tEnv.toDataSet(t, Row.class).output(new DiscardingOutputFormat<Row>());
```

Comment:
This tests doesn't check anything except that it doesn't throw an exception. We should at least count the number of returned records.